### ISSUE TITLE
Сделать, чтобы в сгенерённом проекте работал healthcheck

### ISSUE THREAD

**Issue by f213** [fandsdev/django#576]
Сейчас мы не добавляем healthcheck в итоговый проект, потому что указываем в нём HTTP HOST. Нужно решить эту проблему — к примеру сделать так, чтобы итоговая джанга не требовала правильного хоста — в 2024 году это не актуально.

**Comment by nkiryanov** [comment]
>Сейчас мы не добавляем healthcheck в итоговый проект, потому что указываем в нём HTTP HOST

А можешь плз пояснить плз.
сс @kazqvaizer может быть ты знаешь.

Насколько понимаю в проекте по умолчанию создаем эндпоинт `api/v1/healthchecks/` в котором чекаем коннекшн с БД:
1. [Вот тут](https://github.com/fandsdev/django/blob/fb2c4063ae845b3e4a281872583083deebc6ef1e/%7B%7B%20cookiecutter.name%20%7D%7D/pyproject.toml#L25) устанавливаем `django-healthchecks`
2. [Тут](https://github.com/fandsdev/django/blob/fb2c4063ae845b3e4a281872583083deebc6ef1e/%7B%7B%20cookiecutter.name%20%7D%7D/src/app/conf/healthchecks.py#L6) конфигрурем какой healthcheck использовать, [a тут](https://github.com/fandsdev/django/blob/fb2c4063ae845b3e4a281872583083deebc6ef1e/%7B%7B%20cookiecutter.name%20%7D%7D/src/app/urls/v1.py#L10) добавляем url

Или речь про что-то другое?

p.s.
вообще у меня давно предлложение — а не написать ли healthcheck самим. В 99% тестируем только коннекшн с БД и это оч мало кода. Так смогли бы убрать одну зависимость (к слову django-healthchecks [не обновлялся 3 года](https://github.com/labd/django-healthchecks) и уже сталкивался с ворнингами)

**Comment by kazqvaizer** [comment]
Подозреваю, что речь про HEALTHCHECK инструкцию внутри Dockerfile, для uwsgi её нету. Т.е. что-то вроде

```Dockerfile
HEALTHCHECK --interval=15s --timeout=15s --start-period=5s --retries=3 \
  CMD wget -q -O /dev/null http://localhost:8000/api/v1/healthchecks/postgresql/ --header "Host: MY-HOST.RU" || exit 1
```

**Comment by kazqvaizer** [comment]
> к слову django-healthchecks [не обновлялся 3 года](https://github.com/labd/django-healthchecks) и уже сталкивался с ворнингами

Зачем менять то, что сделано настолько хорошо!

А если серьезно, то согласен, давай обновим, вот, к примеру:

https://github.com/revsys/django-health-check

**Comment by f213** [comment]
@kazqvaizer а почему не хочешь сделать свою реализацию? Это же 5 строк кода, и избавимся от целой зависимости!

```py
from django import AnyStockDjangoModel

def healthcheck(request):
  AnyStockDjangoModel.objects.exists()
  return 'ok'
```

---
Правда тут задача немного про другое — мы сейчас не можем сделать работающий по дефолту healthcheck, потому что в докерфайле должен быть обязательно указан HTTP-хост, который прописан в проекте. Это дыра с точки зрения DX — мы заставляем программиста думать о какой-то неважной фигне: хттп-хостах вместо бизнеса.

Этот хедер приходится добавлять из-за проверки `ALLOWED_HOSTS`, которая наглухо вшита в джангу, и не несёт никакого смысла в 2025 году. Просто так от неё не избавиться — `ALLOWED_HOSTS` нельзя указывать пустым, и единственный вариант, который я придумал — это манкипатчить [get_host()](https://github.com/django/django/blob/1831f7733d3ef03d1ca7fac3e8d9f4c5e3e3375e/django/http/request.py#L166).

**Comment by kazqvaizer** [comment]
> а почему не хочешь сделать свою реализацию?

А я и не против, но пакет выше любопытный. 

В целом согласен, все атаки на ALLOWED_HOST действительно редкие и защищаться от них нет смысла, т.к. для этого нужно написать код, который полагается на Host в заголовке. Я правда, считал, что это отключается ALLOWED_HOST = ['*'], странно, что это не работает и нужно патчить

**Comment by nkiryanov** [comment]
> Этот хедер приходится добавлять из-за проверки ALLOWED_HOSTS, которая наглухо вшита в джангу, и не несёт никакого смысла в 2025 году. Просто так от неё не избавиться — ALLOWED_HOSTS нельзя указывать пустым, и единственный вариант, который я придумал — это манкипатчить [get_host()](https://github.com/django/django/blob/1831f7733d3ef03d1ca7fac3e8d9f4c5e3e3375e/django/http/request.py#L166).

Вариант `ALLOWED_HOST = ['*']` норм работает
Добавляют в #792 , там же скриншот что работает норм (заодно проверял когда ломалось).

Переписать healthcheck вручную — сделаю отдельно.

**Comment by f213** [comment]
> вариант ALLOWED_HOST = ['*'] норм работает

Уверен, что ты проверял с DEBUG=False? Я не могу найти по коду, но точно помню, что где-то на проде у нас `ALLOWED_HOST = ['*'] `не работал.

**Comment by nkiryanov** [comment]
> Уверен, что ты проверял с DEBUG=False? 

Угу — разворачивал проект с нуля + почитал [документацию](https://docs.djangoproject.com/en/5.1/ref/settings/#allowed-hosts). Там `*` валидное значение.
Допускаю конечно, что что-то потерялось, но проверял как в прод разворачивал бы.

**Comment by f213** [comment]
Я можешь на всякий случай проверить, работает ли такая конструкция для не-FQDN хостов? Скажем, овтечает ли нормально прилоежение, если запросить его по адресу `http://app_backend:8000/`? Кажется у меня проблема была где-то в этом районе

**Comment by nkiryanov** [comment]
> Я можешь на всякий случай проверить, работает ли такая конструкция для не-FQDN хостов? 

Да, всё в порядке. Пример: добавил такого клиента в `compose.yml`:

![Image](https://github.com/user-attachments/assets/f61dd718-da41-47c1-a42e-0547dca77f6b)

и в клиенте проверил, что бэкежнд доступен:

![Image](https://github.com/user-attachments/assets/19fdf72c-2e32-462d-895a-fd12b1f7177f)

**Comment by nkiryanov** [comment]
сделали в #792
